apps/src/templates/ProtectedStatefulDiv.jsx

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+/**
+ * A div DOM element that will never update its contents and will throw an
+ * exception if it is ever unmounted, enforcing that it must always be rendered
+ * because its contents may contain state that the application is depending on.
+ *
+ * Useful when React is wrapping external libraries or parts of our UI that are
+ * not yet driven by React.
+ */
+var ProtectedStatefulDiv = React.createClass({
+  propTypes: {
+    id: React.PropTypes.string
+  },
+
+  shouldComponentUpdate: function () {
+    return false;
+  },
+
+  componentWillUnmount: function () {
+    throw new Error("Unmounting a ProtectedStatefulDiv is not allowed.");
+  },
```

Comment:
I thought about that, and have verified that the component does not unmount when navigating away from the page.

I'm tempted to hold off on writing the 'okToUnmount' path until it's actually needed.  Thoughts?